iceoryx_posh/test/moduletests/test_popo_server_port_user.cpp

```diff
@@ -0,0 +1,732 @@
+}
+
+TEST_F(ServerPort_test, HasNewRequestsWithOneRequestIsTrue)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "0235ce49-96b3-41c0-aac1-e0d6f6bc2b1f");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    constexpr uint64_t NUMBER_OF_REQUESTS{1U};
+    pushRequests(sut.requestQueuePusher, NUMBER_OF_REQUESTS);
+
+    EXPECT_TRUE(sut.portUser.hasNewRequests());
+}
+
+TEST_F(ServerPort_test, HasNewRequestsWithNoRequestsButPreviouslyHavingOneIsFalse)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "29b0a347-2439-488d-ba13-962c7acffac1");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    constexpr uint64_t NUMBER_OF_REQUESTS{1U};
+    pushRequests(sut.requestQueuePusher, NUMBER_OF_REQUESTS);
+    IOX_DISCARD_RESULT(sut.portUser.getRequest());
+
+    EXPECT_FALSE(sut.portUser.hasNewRequests());
+}
+
+TEST_F(ServerPort_test, HasNewRequestsWithOneRequestButIntermediatelyHavingNoneIsTrue)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "a9fc0d89-7b97-48c4-8ba4-e4b22355116b");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    constexpr uint64_t NUMBER_OF_REQUESTS{1U};
+    pushRequests(sut.requestQueuePusher, NUMBER_OF_REQUESTS);
+    IOX_DISCARD_RESULT(sut.portUser.getRequest());
+    pushRequests(sut.requestQueuePusher, NUMBER_OF_REQUESTS);
+
+    EXPECT_TRUE(sut.portUser.hasNewRequests());
+}
+
+TEST_F(ServerPort_test, HasNewRequestsWithMultipleRequestsIsTrue)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "30e36e69-4f23-41ee-aac7-d34240f075ae");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    constexpr uint64_t NUMBER_OF_REQUESTS{2U};
+    pushRequests(sut.requestQueuePusher, NUMBER_OF_REQUESTS);
+
+    EXPECT_TRUE(sut.portUser.hasNewRequests());
+}
+
+TEST_F(ServerPort_test, HasNewRequestsWithMultipleRequestsAndAllButOneRemovedIsTrue)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "96c42094-9931-46a3-8fd6-e7d28b490527");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    constexpr uint64_t NUMBER_OF_REQUESTS{2U};
+    pushRequests(sut.requestQueuePusher, NUMBER_OF_REQUESTS);
+    IOX_DISCARD_RESULT(sut.portUser.getRequest());
+
+    EXPECT_TRUE(sut.portUser.hasNewRequests());
+}
+
+TEST_F(ServerPort_test, HasNewRequestsWithNoRequestsButIntermediatellyHavingMultipleRequestsIsFalse)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "44845e16-ec0a-4b4e-a000-ff62f377c0b9");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    constexpr uint64_t NUMBER_OF_REQUESTS{2U};
+    pushRequests(sut.requestQueuePusher, NUMBER_OF_REQUESTS);
+    IOX_DISCARD_RESULT(sut.portUser.getRequest());
+    IOX_DISCARD_RESULT(sut.portUser.getRequest());
+
+    EXPECT_FALSE(sut.portUser.hasNewRequests());
+}
+
+// END hasNewRequests tests
+
+// BEGIN getRequest tests
+
+TEST_F(ServerPort_test, GetRequestWithoutOfferResultsIn_NO_CHUNK_AVAILABLE)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "6c555224-8b03-46c5-a988-059ac2656149");
+    auto& sut = serverPortWithoutOfferOnCreate;
+
+    sut.portUser.getRequest()
```

Comment:
Could we maybe adjust the API here. I would introduce a `ServerPortRequestResult` which contains the `ChunkReceiveResult` values as a subset and adds `SERVER_DOES_NOT_OFFER`. You could also add a translation function which translates the `ChunkReceiveResult` into a `ServerPortRequestResult` so that it can be handled easier.

In this case I would then recommend to return `ServerPortRequestResult::SERVER_DOES_NOT_OFFER` which states the actual reason.

Reply:
I'm a bit torn on this. We also do not do it for `SubscriberPort` when not subscribed and just return a `NO_CHUNK_AVAILABLE`. I guess this is due to the fact that we can be unsubscribed even after calling `subscribe` if there is no publisher. With the `ServerPort` it is a bit different, since the port controls the state by itself and it is indeed binary. I was hesitant about creating a separate error enum since the API wouldn't mirror the pub-sub and client errors but would have an own one.
What do you think after knowing my considerations? I'm happy to change it if you think the more explicit error message outweighs the familiarity of the API.

Reply:
done